File: backend/models/Habit.js
import mongoose from "mongoose";

const habitSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },
    description: {
      type: String,
      default: "",
    },
    frequency: {
      type: String,
      enum: ["daily", "weekly"],
      required: true,
    },
    status: {
      type: String,
      enum: ["active", "archived"],
      default: "active",
    },
    owner: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
  },
  { timestamps: true }
);

export default mongoose.model("Habit", habitSchema);

File: frontend/lib/models/habit.dart
class Habit {
  final String id;
  final String title;
  final String description;
  final String frequency;
  final String status;

  Habit({
    required this.id,
    required this.title,
    required this.description,
    required this.frequency,
    required this.status,
  });

  factory Habit.fromJson(Map<String, dynamic> json) {
    return Habit(
      id: json['_id'],
      title: json['title'],
      description: json['description'] ?? '',
      frequency: json['frequency'],
      status: json['status'],
    );
  }
}